src/components/MapMaplibreGl.tsx

```diff
@@ -87,7 +90,12 @@ class MapMaplibreGlInternal extends React.Component<MapMaplibreGlInternalProps,
     this.state = {
       map: null,
       inspect: null,
+      geocoder: null,
+      zoomControl: null,
     }
+    i18next.on('languageChanged', () => {
```

Comment:
I'm not sure this is the right solution, but I couldn't find a simpler one.
The `shouldComponentUpdate` method is returning false when chaning the language and the `render` method is not called.
I found this solution here:
https://github.com/i18next/react-i18next/issues/831

I'm not sure this is the right solution though...

Reply:
I think this is OK for now. When implementing the i18next, there were some times where I thought that it would have been less complicated to use functional components instead of class components -- things like these will probably be cleaned up when migrating to functional components